sql/item_keyring_func.cc

```diff
@@ -0,0 +1,54 @@
+#include <my_global.h>
+#include "item_keyring_func.h"
+#include <mysql/service_mysql_keyring.h>
+#include <algorithm>
+#include <cstring>
+#include "sql_class.h"           // THD
+#include "system_key.h"
+
+bool Item_func_rotate_system_key::itemize(Parse_context *pc, Item **res)
+{
+  if (skip_itemize(res))
+    return false;
+  if (Item_bool_func::itemize(pc, res))
+    return true;
+  pc->thd->lex->set_stmt_unsafe(LEX::BINLOG_STMT_UNSAFE_UDF);
```

Comment:
Might makes sense to add a STM testcase to check for this warning

Reply:
Done